examples/basic/ipc_shm_ipcapi_producer.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <umf/ipc.h>
+#include <umf/memory_pool.h>
+#include <umf/pools/pool_jemalloc.h>
+#include <umf/providers/provider_os_memory.h>
+
+#define INET_ADDR "127.0.0.1"
+#define MSG_SIZE 1024
+
+int main(int argc, char *argv[]) {
+    struct sockaddr_in consumer_addr;
```

Comment:
Sockets is a platform-specific way perhaps for simplification purposes we can use just files to communicate between processes. The producer creates an IPC handle and writes it to the file and the consumer process reads it from the file. Such example will be cross-platform.

Reply:
I am afraid it would not be simpler. `recv()` waits for a message. We would have to implement a mechanism of waiting for a complete message being written into the file (checking if the message was written to the file and if it is complete).

Reply:
I think that since we already have this example working we could merge this version. As for now, we don't have plans to use IPC on Windows. OS provider IPC is also implemented on Linux only, on Windows it would return the NOT_SUPPORTED error for now.